src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,146 @@
+        User newUser = createUser(SHORT_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(SHORT_LOGIN));
+        assertThrows(RegistrationException.class, () -> registrationValidator.register(newUser));
+        assertNull(storageDao.get(SHORT_LOGIN));
+    }
+
+    @Test
+    public void register_nullLogin_notOk() {
+        User newUser = createUser(null, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(null));
+        assertThrows(RegistrationException.class, () -> registrationValidator.register(newUser));
+        assertNull(storageDao.get(null));
+    }
+
+    @Test
+    public void register_emptyLogin_notOk() {
+        User newUser = createUser(EMPTY_STRING, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(EMPTY_STRING));
+        assertThrows(RegistrationException.class, () -> registrationValidator.register(newUser));
+        assertNull(storageDao.get(EMPTY_STRING));
+    }
+
+    @Test
+    public void register_validPassword_ok() throws RegistrationException {
+        User newUser = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(VALID_LOGIN));
+        registrationValidator.register(newUser);
+        assertEquals(newUser, storageDao.get(VALID_LOGIN));
+    }
+
+    @Test
+    public void register_invalidPassword_notOk() {
+        User newUser = createUser(VALID_LOGIN, SHORT_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(VALID_LOGIN));
+        assertThrows(RegistrationException.class, () -> registrationValidator.register(newUser));
+        assertNull(storageDao.get(VALID_LOGIN));
+    }
+
+    @Test
+    public void register_nullPassword_notOk() {
+        User newUser = createUser(VALID_LOGIN, null, VALID_AGE);
+        assertNull(storageDao.get(VALID_LOGIN));
+        assertThrows(RegistrationException.class, () -> registrationValidator.register(newUser));
+        assertNull(storageDao.get(VALID_LOGIN));
+    }
+
+    @Test
+    public void register_emptyPassword_notOk() {
+        User newUser = createUser(VALID_LOGIN, EMPTY_STRING, VALID_AGE);
+        assertNull(storageDao.get(VALID_LOGIN));
+        assertThrows(RegistrationException.class, () -> registrationValidator.register(newUser));
+        assertNull(storageDao.get(VALID_LOGIN));
+    }
+
+    @Test
+    public void register_validAge_ok() throws RegistrationException {
+        User newUser = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertNull(storageDao.get(VALID_LOGIN));
+        registrationValidator.register(newUser);
+        assertEquals(newUser, storageDao.get(VALID_LOGIN));
+    }
+
+    @Test
+    public void register_invalidAge_notOk() {
+        User newUser = createUser(VALID_LOGIN, VALID_PASSWORD, INVALID_AGE);
+        assertNull(storageDao.get(VALID_LOGIN));
+        assertThrows(RegistrationException.class, () -> registrationValidator.register(newUser));
+        assertNull(storageDao.get(VALID_LOGIN));
```

Comment:
The test `register_invalidAge_notOk` is checking for an age that is too young, but the name `INVALID_AGE` is not descriptive enough. Consider renaming it to `UNDERAGE` to better reflect the intent of the test.